include/boost/math/special_functions/rsqrt.hpp

```diff
@@ -15,7 +15,7 @@ template<typename Real>
 inline Real rsqrt(Real const & x)
 {
     using std::sqrt;
-    if constexpr (std::is_same_v<Real, float> || std::is_same_v<Real, double> || std::is_same_v<Real, long double>)
+    if constexpr (std::is_floating_point_v<Real>)
```

Comment:
@NAThompson : was the original intention here to exclude `__float128`?  If so replacing with is_floating_point *might* fail in -std=gnu++ mode.   Just double checking...

Reply:
Amazingly enough I remember this! Yes, that was the goal.

As you can see in the `else` branch, the `rsqrt` iteration starts with the `long double` approximation and then refines it with the Newton method for `1/sqrt(x)`. I verified that it is slower to call `1/sqrt(x)` for `float128` than to use the `long double` approximation and Newton iterate.